### PR TITLE
Webhook Response Enhancement

### LINKED ISSUE #186: Webhook output in JSON format
I have configure kubewatch to send notification in webhook.

I Webhook I am getting in flat text.

```{"text":"A `pod` in namespace `sample-app` has been `Deleted`:\n`sample-app/yelb-ui-5946c9cfdb-gqpbb`"}```

Not sure its supported JSON or not but if you could do JSON output enhancement as below in webhook that will be good.

```
{
   "output" : "A `pod` in namespace `sample-app` has been `Deleted`:\n`sample-app/yelb-ui-5946c9cfdb-gqpbb`"
   "time" : "2017-10-09T23:31:56",
   "output_fields" : {
      "pod" : "yelb-ui-5946c9cfdb-gqpbb",
      "namespace" : "sample-app",
      "action" : "Deleted"
   }
}
```

### PR DESCRIPTION
**Pull request by codenio**
This PR address #186
- Enhances the webhook JSON Payload
- Adds EventMeta, Message, Time fields to JSON response.

changes the current JSON payload 

from
```
{
"text" : "A `pod` in namespace `sample-app` has been `Deleted`:\n`sample-app/yelb-ui-5946c9cfdb-gqpbb`"
}
```
to
```JSON
{
  "eventmeta": {
    "kind": "object",
    "name": "Test",
    "namespace": "default",
    "reason": "Tested"
  },
  "message": "Testing Handler Configuration. This is a Test message.",
  "time": "2019-07-18T15:29:48.677368507+05:30"
}
```

Comments and feedbacks..?

Closes #186

### CODE REVIEW

**Comment by prasenforu** [comment]
Great !!

Let me know when image are ready...

Once again thank you.

**Comment by aananthraj** [comment]
(for the time being) docker image with the latest code changes is available at [cipherco/kubewatch](https://cloud.docker.com/u/cipherco/repository/docker/cipherco/kubewatch)

**Comment by prasenforu** [comment]
@aananthraj 

Thanks for image.

Looks like I am not getting right (original) event ```reason``` like image pull, all are customised as "Updated"

lLive watch for 1 hr

https://www.webhook.site/#!/8ce3260e-ffb0-4a80-aed0-bf8ec30fed33/c217ba81-5088-43c6-bf5e-109d883a14f7/1

**Comment by aananthraj** [comment]
@prasenforu available reasons are : Created, Updated, Deleted.

**Comment by prasenforu** [comment]
But original reason is important specially for pod which helps to diagnose.

**Comment by aananthraj** [comment]
presently the exact reason for the creation of Update events is not being captured. (which makes kubewatch niosy) 

the latest code (available in cipher.co/kubewatch) provides us an option to skip such noisy update events using the Events config section in the config file

Using Event Config:
```
$ cat ~/.kubewatch.yaml 
handler:
  slack:
    token: xoxb-xxxxx-yyyyyyy
    channel: kube-watch-test

event:
    global:                       // global section alerts for all events
     - pod
     - deployment
    create:                       // create section alerts for resource object creation
     - service
    update:                       // update section for resource object updation
     - 
    delete:                       // delete section alerts for resource object deletion
     - job
     - service

namespace: ""
```

**Comment by prasenforu** [comment]
If noise help us to go right direction then such noise always acceptable instead of dummy message.

File: pkg/handlers/webhook/webhook.go [hunk 1]
```diff
@@ -50,7 +50,17 @@ type Webhook struct {
 
 // WebhookMessage for messages
 type WebhookMessage struct {
-	Text string `json:"text"`
+	EventMeta EventMeta `json:"eventmeta"`
+	Message   string    `json:"message"`
```
Comment (CKarper) [inline pkg/handlers/webhook/webhook.go]: This should remain as 'Text` so it doesn't break existing webhooks.
```suggestion
	Text string `json:"text"`
```

Comment (aananthraj) [reply pkg/handlers/webhook/webhook.go]: resolved

**Comment by prasenforu** [comment]
@aananthraj

Not sure image is ready or not.

But I was playing with cipherco/kubewatch
 And recently I observed that some abnormal behaviour.

With POD creation event not showing proper namespace.  

Sometimes it's taking namespace where kubewatch is running even all update events also taking same where kubewatch is running.

**Comment by aananthraj** [comment]
@prasenforu can we create a issue under cipher.co/kubewatch with further details. So that we can work on it and fix them

**Comment by prasenforu** [comment]
@aananthraj 
Not able to view Issue tab to create am issue ..

image

**Comment by aananthraj** [comment]
@prasenforu , now it is made available, sorry for that.

**Comment by codenio** [comment]
FYI, These features are now avalilable with [Botkube ](https://github.com/infracloudio/botkube) (0.9.0)
People in need can make use of it. :+1:

**Comment by mkmik** [comment]
thanks; sorry for the slowness; we've got a bit distracted after the acquisition; will try to clear up the backlog
and make this tool useful again

**Comment by mkmik** [comment]
bors&github, you're drunk; merging manually

**Pull request merged**